engo_glfw.go

```diff
@@ -217,20 +224,44 @@ func RunPreparation(defaultScene Scene) {
 	WorldBounds.Max = Point{Width(), Height()}
 
 	SetScene(defaultScene, false)
+	game = defaultScene
+}
+
+func closeEvent() {
+	log.Println("Got close event")
+	game.Exit()
+	if defaultCloseAction {
+		log.Println("Closing")
+		Exit()
+	} else {
+		log.Println("Warning: default close action set to false, please make sure you manually handle this")
+	}
 }
 
 func runLoop(defaultScene Scene, headless bool) {
+	c := make(chan os.Signal, 1)
+    signal.Notify(c, os.Interrupt)
+    signal.Notify(c, syscall.SIGTERM)
+    go func() {
+        <-c
+		closeEvent()
+    }()
+
 	RunPreparation(defaultScene)
 
 	ticker := time.NewTicker(time.Duration(int(time.Second) / fpsLimit))
-Outer:
-	for {
+	Outer: for {
 		select {
 		case <-ticker.C:
 			RunIteration()
-			if !headless && window.ShouldClose() {
+			if close {
```

Comment:
@EtienneBruines no, close is set when the game is going to be closed. ShouldClose and the signal call closeEvent().